File: ans/q01_35/q05/q05.py
count = 0
coins = [0, 10, 50, 100, 500]
for i in coins:
    for j in coins:
        if j > i:
            break
        for k in coins:
            if k > j:
                break
            for l in coins:
                if l > k:
                    break
                for m in coins:
                    if m > l:
                        break
                    for n in coins:
                        if n > m:
                            break
                        for o in coins:
                            if o > n:
                                break
                            for p in coins:
                                if p > o:
                                    break
                                for q in coins:
                                    if q > p:
                                        break
                                    for r in coins:
                                        if r > q:
                                            break
                                        for s in coins:
                                            if s > r:
                                                break
                                            for t in coins:
                                                if t > s:
                                                    break
                                                for u in coins:
                                                    if u > t:
                                                        break
                                                    for v in coins:
                                                        if v > u:
                                                            break
                                                        for w in coins:
                                                            if w > v:
                                                                break
                                                            if i+j+k+l+m+n+o+p+q+r+s+t+u+v+w == 1000:
                                                                count += 1
                                                                # print(i, j, k, l, m, n, o, p, q, r, s, t, u, v, w)
print(count)

File: ans/q01_35/q05/q05.cpp
// author kbb
#include <iostream>
using namespace std;

int main()
{
    int sum_coins;//枚数
    int max_coins;//値段
    int count = 0;
        for(int coin500 = 0; coin500 <= 2; coin500++){
            for(int coin100 = 0; coin100 <= 10; coin100++){
                for(int coin50 = 0; coin50 <= 15; coin50++){
                    for(int coin10 = 0; coin10 <= 15; coin10++){
                        sum_coins = coin500 + coin100 + coin50 + coin10;
                        max_coins = coin500*500 + coin100*100 + coin50*50 + coin10*10;
                        if(sum_coins <= 15 && max_coins == 1000){
                            count++;
                        }
                    }
                }
            }
        }
        cout<<count<<endl;
}
